data-entry/dataEntry.app.js

```diff
@@ -74,23 +74,80 @@
+                                    // no vocab annotation, do nothing
+                                }
+
+                                try {
+                                    var displayAnnotation = ftable.annotations.get("tag:misd.isi.edu,2015:display");
+                                } catch (error) {
+                                    // no display annotation, do nothing
+                                }
+
+                                if (vocabAnnotation) {
+                                    if (vocabAnnotation.content.term) {
+                                        var termColumn = ftable.columns.get(vocabAnnotation.content.term);
+                                        displayColumns.push(termColumn); // the array is now [keyColumn, termColumn]
+                                    }
+                                    // vocabulary term is undefined
+                                    else {
+                                        var ftableColumns = ftable.columns.all();
+                                        for (var i = 0, length = ftableColumns.length; i < length; i++) {
+                                            var uppColumnName = $filter('uppercase')(column.name);
+                                            if (uppColumnName == 'TERM' || uppColumnName == 'NAME') {
+                                                displayColumns.push(column);
+                                                break;
+                                            }
+                                        } /* term undefined */
+
+                                        pattern = "{" + displayColumns[1].name + "}";
+                                    }
+                                    /* END USE CASE 2 */
+                                }
+                                /* THIRD USE CASE: not a vocabulary but it has a “display : row name” annotation */
+                                else if (displayAnnotation) {
+                                    if (displayAnnotation.content.row_name) {
+                                        // TODO
+                                        // var array_of_col_names = REGEX THE array of column_name strings from “ … `{` column_name `}` …” patterns
+                                        // angular.forEach(array_of_col_names, function(column_name) {
+                                        //     displayColumns.push(table.columns.get(column_name));
+                                        // });
+                                        //
+                                        // pattern = displayAnnotation.row_name;
+                                    }
+                                }
+                            } finally {
+                                try {
+                                    (function(fkey) {
+                                        ftable.entity.get(null, null, displayColumns).then(function success(columns){
```

Comment:
I get why you called the returned data `columns` and it made sense after I thought about it for a second, but it confused me at first because that term overlaps in my mind with the notion of columns from our schema. I think it would be better to call it `rowset` since it will be an instance of the `RowSet` class.